tests/test_flats.py

```diff
@@ -0,0 +1,50 @@
+"""
+Unit test code for flat production code.
+"""
+from collections import namedtuple
+import unittest
+import numpy as np
+import galsim
+import desc.imsim
+
+TreeRingInfo = namedtuple('TreeRingInfo', ['center', 'func'])
+
+class GsDetector:
+    """
+    Minimal implementation of an interface-compatible version
+    of GalSimDetector for testing the flat production code.
+    """
+    def __init__(self):
+        self.xMaxPix = 256
+        self.yMaxPix = 256
+        self.xMinPix = 1
+        self.yMinPix = 1
+        self.wcs = galsim.wcs.PixelScale(0.2)
+        self.tree_rings \
+            = TreeRingInfo(galsim.PositionD(0, 0),
+                           galsim.SiliconSensor.simple_treerings(0.26, 47))
+
+
+class FlatTestCase(unittest.TestCase):
+    """TestCase class for flat production code."""
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def test_make_flat(self):
+        """Test of make_flat function."""
+        gs_det = GsDetector()
+        counts_per_iter = 1000
+        niter = 10
+        rng = galsim.UniformDeviate()
+        logger = desc.imsim.get_logger('WARN')
+        flat = desc.imsim.make_flat(gs_det, counts_per_iter, niter, rng,
+                                    logger=logger)
+        self.assertLess(np.abs(niter*counts_per_iter - flat.array.mean()),
```

Comment:
I'm not quite sure I understand what is being tested here and how it relates to the tree rings model.  I think this is just checking that the overall level of the flat is correct and the tree rings are irrelevant.  So mostly this is just checking the flat level itself.  I think a comment saying what is being checked would be useful.  The thing I'm not sure about is how the variance caused by the addition of the treerings changes the std().

Reply:
This is mostly an operational test that the flats had the expected level.  Using stdev isn't very stringent (and not the test to use if one were really concerned about the statistical properties of the pixel values, which we aren't here), but it's convenient since I was playing around with image sizes and total counts per pixel for the test and wanted something that would adjust for all that automatically.  In any case, this test code is not testing the treering implementation; it's testing the  flat generation.